nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1330 @@
+        - names like left_cortex, cortex_left, LeftCortex, or CortexLeft will be converted to
+          CIFTI_STRUCTURE_CORTEX_LEFT
+
+        see :py:func:`nibabel.cifti2.tests.test_name` for examples of
+        which conversions are possible
+
+        Parameters
+        ----------
+        name: (str, tuple)
+            input name of an anatomical region
+
+        Returns
+        -------
+        CIFTI2 compatible name
+
+        Raises
+        ------
+        ValueError: raised if the input name does not match a known anatomical structure in CIFTI
+        """
+        if name in cifti2.CIFTI_BRAIN_STRUCTURES:
+            return name
+        if not isinstance(name, string_types):
+            if len(name) == 1:
+                structure = name[0]
+                orientation = 'both'
+            else:
+                structure, orientation = name
+                if structure.lower() in ('left', 'right', 'both'):
+                    orientation, structure = name
+        else:
+            orient_names = ('left', 'right', 'both')
+            for poss_orient in orient_names:
+                idx = len(poss_orient)
+                if poss_orient == name.lower()[:idx]:
+                    orientation = poss_orient
+                    if name[idx] in '_ ':
+                        structure = name[idx + 1:]
+                    else:
+                        structure = name[idx:]
+                    break
+                if poss_orient == name.lower()[-idx:]:
+                    orientation = poss_orient
+                    if name[-idx - 1] in '_ ':
+                        structure = name[:-idx - 1]
+                    else:
+                        structure = name[:-idx]
+                    break
+            else:
+                orientation = 'both'
+                structure = name
+        if orientation.lower() == 'both':
+            proposed_name = 'CIFTI_STRUCTURE_%s' % structure.upper()
+        else:
+            proposed_name = 'CIFTI_STRUCTURE_%s_%s' % (structure.upper(), orientation.upper())
+        if proposed_name not in cifti2.CIFTI_BRAIN_STRUCTURES:
+            raise ValueError('%s was interpreted as %s, which is not a valid CIFTI brain structure'
+                             % (name, proposed_name))
+        return proposed_name
+
+    @property
+    def is_surface(self, ):
+        """
+        (N, ) boolean array which is true for any element on the surface
+        """
+        return np.vectorize(lambda name: name in self.nvertices.keys())(self.name)
+
+    _affine = None
+
+    @property
+    def affine(self, ):
+        """
+        Affine of the volumetric image in which the greyordinate voxels were defined
+        """
+        return self._affine
+
+    @affine.setter
+    def affine(self, value):
+        if value is not None:
+            value = np.asarray(value)
+            if value.shape != (4, 4):
+                raise ValueError('Affine transformation should be a 4x4 array')
```

Comment:
There are a lot of validation checks like this that don't get exercised in the tests. Could you add them?